Lombiq.Tests.UI/Extensions/ShortcutsUITestContextExtensions.cs

```diff
@@ -431,74 +437,113 @@ public interface IShortcutsApi
+        bool activateShell = true) =>
+            context.Application.UsingScopeAsync(
+                async serviceProvider =>
                 {
-                    throw new ThemeNotFoundException($"{id} not found.");
-                }
+                    var shellFeatureManager = serviceProvider.GetRequiredService<IShellFeaturesManager>();
+                    var themeFeature = (await shellFeatureManager.GetAvailableFeaturesAsync())
+                        .FirstOrDefault(feature => feature.IsTheme() && feature.Id == id);
 
-                if (IsAdminTheme(themeFeature.Extension.Manifest))
-                {
-                    var adminThemeService = shellScope.ServiceProvider.GetRequiredService<IAdminThemeService>();
-                    await adminThemeService.SetAdminThemeAsync(id);
-                }
-                else
-                {
-                    var siteThemeService = shellScope.ServiceProvider.GetRequiredService<ISiteThemeService>();
-                    await siteThemeService.SetSiteThemeAsync(id);
-                }
+                    if (themeFeature == null)
+                    {
+                        throw new ThemeNotFoundException($"Theme with the feature ID {id} not found.");
+                    }
 
-                var enabledFeatures = await shellFeatureManager.GetEnabledFeaturesAsync();
-                var isEnabled = enabledFeatures.Any(feature => feature.Extension.Id == themeFeature.Id);
+                    if (IsAdminTheme(themeFeature.Extension.Manifest))
+                    {
+                        var adminThemeService = serviceProvider.GetRequiredService<IAdminThemeService>();
+                        await adminThemeService.SetAdminThemeAsync(id);
+                    }
+                    else
+                    {
+                        var siteThemeService = serviceProvider.GetRequiredService<ISiteThemeService>();
+                        await siteThemeService.SetSiteThemeAsync(id);
+                    }
 
-                if (!isEnabled)
-                {
-                    await shellFeatureManager.EnableFeaturesAsync(new[] { themeFeature }, force: true);
-                }
-            },
-            tenant,
-            activateShell);
+                    var enabledFeatures = await shellFeatureManager.GetEnabledFeaturesAsync();
+                    var isEnabled = enabledFeatures.Any(feature => feature.Extension.Id == themeFeature.Id);
+
+                    if (!isEnabled)
+                    {
+                        await shellFeatureManager.EnableFeaturesAsync(new[] { themeFeature }, force: true);
+                    }
+                },
+                tenant,
+                activateShell);
 
     /// <summary>
-    /// Creates, sets up and navigates to a new URL prefixed tenant. Also changes <see cref="UITestContext.TenantName"/>.
+    /// Creates, sets up, switches to, and navigates to a new URL-prefixed tenant.
```

Comment:
It still changes `TenantName` (and the others) via `context.SwitchCurrentTenant()`. I think this should be mentioned at least in a `<remarks>` to avoid confusion as the context's internal state has been changed.